### PR TITLE
feat: SCW allow pause cluster

### PR DESCRIPTION
**Pull request by benjaminch**

### CODE REVIEW

File: src/cloud_provider/scaleway/kubernetes/mod.rs [hunk 1]
```diff
@@ -578,11 +580,195 @@ impl<'a> Kapsule<'a> {
...
+            listeners_helper.pause_in_progress(ProgressInfo::new(
+                ProgressScope::Infrastructure {
+                    execution_id: self.context.execution_id().to_string(),
+                },
+                ProgressLevel::Info,
+                Some(message),
+                self.context.execution_id(),
+            ))
+        };
+        send_to_customer(format!("Preparing SCW {} cluster pause with id {}", self.name(), self.id()).as_str());
```
Comment (MacLikorne) [inline src/cloud_provider/scaleway/kubernetes/mod.rs:594]: there is now a self.send_to_customer method

File: src/cloud_provider/scaleway/kubernetes/mod.rs [hunk 2]
```diff
@@ -578,11 +580,195 @@ impl<'a> Kapsule<'a> {
...
+
+        let temp_dir = workspace_directory(
```
Comment (MacLikorne) [inline src/cloud_provider/scaleway/kubernetes/mod.rs:596]: there is now a selfget_temp_dir method

File: src/cloud_provider/scaleway/kubernetes/mod.rs (hunk 1 above)
Comment (benjaminch) [reply src/cloud_provider/scaleway/kubernetes/mod.rs]: Done

File: src/cloud_provider/scaleway/kubernetes/mod.rs (hunk 2 above)
Comment (benjaminch) [reply src/cloud_provider/scaleway/kubernetes/mod.rs:583]: Done

File: src/cloud_provider/scaleway/kubernetes/mod.rs [hunk 3]
```diff
@@ -652,12 +654,189 @@ impl<'a> Kubernetes for Kapsule<'a> {
...
+                scope: EngineErrorScope::Kubernetes(self.id.clone(), self.name.clone()),
+                execution_id: self.context.execution_id().to_string(),
+                message: Some("No worker nodes present, can't Pause the infrastructure. This can happen if there where a manual operations on the workers or the infrastructure is already pause.".to_string()),
+            });
+        }
+
+        let kubernetes_config_file_path = self.config_file_path()?;
+
+        // pause: wait 1h for the engine to have 0 running jobs before pausing and avoid getting unreleased lock (from helm or terraform for example)
+        let metric_name = "taskmanager_nb_running_tasks";
```
[the remark this answers is not in the document]
Comment (benjaminch) [reply src/cloud_provider/scaleway/kubernetes/mod.rs]: Done

File: src/cloud_provider/digitalocean/kubernetes/mod.rs [hunk 4]
```diff
@@ -828,11 +832,186 @@ impl<'a> Kubernetes for DOKS<'a> {
...
+                scope: EngineErrorScope::Kubernetes(self.id.clone(), self.name.clone()),
+                execution_id: self.context.execution_id().to_string(),
+                message: Some("No worker nodes present, can't Pause the infrastructure. This can happen if there where a manual operations on the workers or the infrastructure is already pause.".to_string()),
+            });
+        }
+
+        let kubernetes_config_file_path = self.config_file_path()?;
+
+        // pause: wait 1h for the engine to have 0 running jobs before pausing and avoid getting unreleased lock (from helm or terraform for example)
+        let metric_name = "taskmanager_nb_running_tasks";
```
[the remark this answers is not in the document]
Comment (benjaminch) [reply src/cloud_provider/digitalocean/kubernetes/mod.rs]: Done

File: src/cloud_provider/scaleway/kubernetes/mod.rs (hunk 1 above)
Comment (benjaminch) [reply src/cloud_provider/scaleway/kubernetes/mod.rs]: Changed in AWS and DO as well

File: src/cloud_provider/scaleway/kubernetes/mod.rs (hunk 2 above)
Comment (benjaminch) [reply src/cloud_provider/scaleway/kubernetes/mod.rs]: Changed in AWS and DO as well

**Review by MacLikorne — APPROVED** [review]

**Pull request merged**